Assets/Scripts/DynamicCamera/CameraBehavior.cs

```diff
@@ -0,0 +1,76 @@
+﻿namespace HarmonyQuest.DynamicCamera
+{
+    using UnityEngine;
+
+    public abstract class CameraBehavior : MonoBehaviour
+    {
+        [SerializeField]
+        private TestPlayer player;
+        private CharacterController characterController;
+        
+        protected Vector3 targetAngles;
+
+        protected float bias = 1f;
+        protected bool active = false;
+        [SerializeField]
+        protected Vector3 direction;
+
+        protected void Init()
+        {
+            characterController = player.GetComponent<CharacterController>();
+        }
+
+        public void Move()
+        {
+            if (!active)
+                return;
+            if (!Vector3.Equals(direction, Vector3.zero))
+            {
+                transform.position = Vector3.Lerp(transform.position, direction, bias * Time.deltaTime);
+            }
+            transform.eulerAngles = Vector3.Lerp(transform.eulerAngles, targetAngles, bias * Time.deltaTime);
+        }
+
+        public void ToggleActive()
+        {
+            active = !active;
+        }
+
+        
+        protected Vector3 PlayerLocation()
+        {
+            return player.transform.position;
+        }
+
+        protected Vector3 PlayerVelocity()
+        {
+            if (!IsFollowingPOI())
+                return characterController.velocity;
+            else
+                return Vector3.zero;
+        }
+
+        protected Vector3 TargetLocation()
+        {
+            return player.lockOnTarget.transform.position;
+        }
+
+        protected bool IsLockedOn()
```

Comment:
Should this be removed?

Reply:
Yeah, for now, though I should add it back once we have the player data in a singleton

Reply:
Also `IsLockedOn()` is redundant because we can just check if `TargetLocation()` is null